### PR TITLE
Update linter, typings and refactor country usage

### PR DESCRIPTION
**Pull request by wdsrocha**
#### What problem is this solving?

Just some minor things to improve `place-components` code quality.

#### How should this be manually tested?

You can test the updated components in [this workspace](https://postalcode--checkoutio.myvtex.com/cart). It might be tricky to find bugs that way, so I suggest to focus more in the code itself.

### CODE REVIEW

File: react/PlaceDetails.tsx [hunk 1]
```diff
@@ -40,31 +43,37 @@ const PlaceDetails: React.FC<Props> = ({
...
+              !hiddenFields.includes(fragment.name as FieldName)
           )
         })
         .filter((line) => line.length > 0)
         .map((line, displayIndex) => (
           <div className="flex" key={displayIndex}>
             {line.map((fragment, index) => {
+              if (typeof index !== 'number') {
+                return
+              }
```
Comment (wdsrocha) [inline react/PlaceDetails.tsx:58]: I don't know why this is needed... index is clearly a `number` already, but eslint complains with [this](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-plus-operands.md) anyway.

File: react/formatAddressToString.ts [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
+      line.filter(({ name }) => isAddressField(name) && address[name])
+    )
     .filter((line) => line.length > 0)
     .map((line) =>
       line
         .map((fragment, index) => {
-          const hasPreviousFragment = index > 0 && address[line[index - 1].name]
+          if (typeof index !== 'number') {
+            return
+          }
```
Comment (wdsrocha) [inline react/formatAddressToString.ts:28]: :ditto:

File: react/formatAddressToString.ts [hunk 3]
```diff
@@ -1,37 +1,50 @@
...
+      line.filter(({ name }) => isAddressField(name) && address[name])
+    )
+    .filter((line) => line.length > 0)
+    .map((line) =>
       line
         .map((fragment, index) => {
-          const hasPreviousFragment = index > 0 && address[line[index - 1].name]
+          if (typeof index !== 'number') {
+            return
+          }
```
Comment (lucasecdb) [inline react/formatAddressToString.ts:28]: how is this not a number? 🤔

File: react/useCountry.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+
+export const useCountry = () => {
+  const { culture } = useRuntime()
+  const { address } = useAddressContext()
+
+  // TODO: should also try to get country from:
+  // - addresses in account
+  // - IP
+  // - store config
```
Comment (wdsrocha) [inline react/useCountry.ts:11]: This comments already existed and I just bring them here. Does anyone knows if we have jira tasks for those already?

File: react/formatAddressToString.ts (hunk 3 above)
Comment (wdsrocha) [reply react/formatAddressToString.ts:28]: ikr lol: https://github.com/vtex-apps/place-components/pull/46#discussion_r582963879

Comment (lucasecdb) [reply react/formatAddressToString.ts:28]: that's weird, maybe something is messed up in our tsconfig..

File: react/useCountry.ts (hunk 4 above)
Comment (lucasecdb) [reply react/useCountry.ts:11]: we do not, also I think "store config" is the one we get from `culture.country` already

File: react/formatAddressToString.ts (hunk 3 above)
Comment (lucasecdb) [reply react/formatAddressToString.ts:28]: can you add a comment above this if so we know we should remove it in the future (if we find a solution)?

Comment (wdsrocha) [reply react/formatAddressToString.ts:28]: sure

Comment (wdsrocha) [reply react/formatAddressToString.ts]: for some reason it's fixed ¯\_(ツ)_/¯
I'm clueless why that was happening but I'll just accept it

**Review by jeffersontuc — APPROVED** [review]
Nice job!

File: react/formatAddressToString.ts (hunk 3 above)
Comment (lucasecdb) [reply react/formatAddressToString.ts]: oh, better yet then 😄

**Review by lucasecdb — APPROVED** [review]

**Pull request merged**